pkg/storage/etcd/etcd_helper_test.go

```diff
@@ -535,3 +559,79 @@ func TestPrefixEtcdKey(t *testing.T) {
 
 	assert.Equal(t, keyBefore, keyAfter, "Prefix incorrectly added by EtcdHelper")
 }
+
+func TestDeleteUIDMismatch(t *testing.T) {
+	server := etcdtesting.NewEtcdTestClientServer(t)
+	defer server.Terminate(t)
+	prefix := path.Join("/", etcdtest.PathPrefix())
+	helper := newEtcdHelper(server.Client, testapi.Default.Codec(), prefix)
+
+	obj := &api.Pod{ObjectMeta: api.ObjectMeta{Name: "foo", UID: "A"}}
+	podPtr := &api.Pod{}
+	err := helper.Create(context.TODO(), "/some/key", obj, podPtr, 0)
+	if err != nil {
+		t.Fatalf("Unexpected error %#v", err)
+	}
+	err = helper.Delete(context.TODO(), "/some/key", obj, api.NewPreconditionsWithUID("B"))
+	err = etcderr.InterpretDeleteError(err, unversioned.GroupResource{}, "")
+	if !errors.IsConflict(err) {
+		t.Fatalf("Expect a Conflict error, got: %v", err)
+	}
+}
+
+type getFunc func(ctx context.Context, key string, opts *etcd.GetOptions) (*etcd.Response, error)
+
+type FakeDeleteKeysAPI struct {
+	etcd.KeysAPI
+	FakeGetFunc getFunc
+	GetCount    *int
+	FakeGetCap  int
```

Comment:
Chcange to desiredNumberOfFakeGets, or add some comments explaining what this will do